src/linear_algebra/dual.lean

```diff
@@ -558,3 +558,100 @@ end
 end
 
 end subspace
+
+variables {R : Type*} [comm_ring R] {M₁ : Type*} {M₂ : Type*}
+variables [add_comm_group M₁] [module R M₁] [add_comm_group M₂] [module R M₂]
+
+open module
+
+/-- Given a linear map `f : M₁ →ₗ[R] M₂`, `f.dual_map` is the linear map between the dual of
+`M₂` and `M₁` such that it maps the functional `φ` to `φ ∘ f`. -/
+def linear_map.dual_map (f : M₁ →ₗ[R] M₂) : dual R M₂ →ₗ[R] dual R M₁ :=
```

Comment:
A lemma that says `linear_map.dual_map linear_map.id = linear_map.id` (with whatever the arguments to `linear_map.id` are) would be good. It's probably true by `rfl` or `funext $ \lam _, rfl`

Reply:
And the marginally more complicated one, `f.dual_map (g.dual_map φ) = (g.comp f).dual_map φ` (or perhaps with f and g swapped)